### PR TITLE
Allow 32-bit build

### PR DESCRIPTION
**Pull request by xantares**
### Description

allows to build in 32 bits windows mode

### Motivation and Context

fixes 32 bits build

### How Has This Been Tested?

compilation in 32 bits mode

### Types of changes

new feature

### Checklist:

- [x] The code has been tested.

### CODE REVIEW

File: contrib/elftoolchain/libelf/_libelf_config.h [hunk 1]
```diff
@@ -157,12 +157,22 @@
...
 #define	LIBELF_BYTEORDER	ELFDATA2LSB
 #define	LIBELF_CLASS		ELFCLASS64
 
+#else
+
+#define LIBELF_ARCH             EM_386
+#define LIBELF_BYTEORDER        ELFDATA2LSB
+#define LIBELF_CLASS            ELFCLASS32
+
+#endif
```
Comment (jpark37) [inline contrib/elftoolchain/libelf/_libelf_config.h:174]: How about this?
```
#ifdef _WIN32

#if	defined(_M_X64)

#define	LIBELF_ARCH		EM_X86_64
#define	LIBELF_BYTEORDER	ELFDATA2LSB
#define	LIBELF_CLASS		ELFCLASS64

#elif	defined(_M_IX86)

#define	LIBELF_ARCH		EM_386
#define	LIBELF_BYTEORDER	ELFDATA2LSB
#define	LIBELF_CLASS		ELFCLASS32

#else
#error	Unknown Windows architecture.
#endif
#endif  /* _WIN32 */
```

**Review by jpark37 — CHANGES_REQUESTED** [review]

**Comment by jpark37** [comment]
Can you update the description? What is this for? Do you really need to use ELF? My impression is that the Mesa maintainers would like to rid themselves of this dependency.

**Comment by xantares** [comment]
I dont know whats its actually used for, but they did not remove it yet.

**Comment by jpark37** [comment]
I have added a description template to the PR. Please fill it out if you want me to merge the changes.

**Comment by jpark37** [comment]
What are your Mesa Meson build arguments?

**Comment by xantares** [comment]
I enabled the amd vulkan driver

**Comment by jpark37** [comment]
Okay, I'm closing the PR.

If anyone related to RADV has a need to build the driver for 32-bit Windows, then I'll reconsider. I'm not interested in testing and merging PRs for this library unless they serve Mesa in some useful way.

**Comment by xantares** [comment]
how that's not what I'm doing ?
please reconsider

**Pull request closed without merging**